### ISSUE TITLE
[Bug]: pdf generated without images (canvas)

### ISSUE THREAD

**Issue by a-apiko** [puppeteer/puppeteer#10341]
### Bug expectation

Expected: pdf file generated with images (canvas)
Actual: pdf file generated without images on first run

Also I found workaround for it. We need to run **screenshot** function and then run **pdf()** once again
After this manipulation in second pdf all images are loaded, if you skip **screenshot()** function not all images will be loaded
Writing this workaround as it may be helpful to understand what's going wrong here or what events triggered and cause this issue to be fixed in this way. 
[1_pdf.pdf](https://github.com/puppeteer/puppeteer/files/11676290/1_pdf.pdf)
[2_pdf.pdf](https://github.com/puppeteer/puppeteer/files/11676292/2_pdf.pdf)

### Bug behavior

- [ ] Flaky
- [X] PDF

### Minimal, reproducible example

```TypeScript
const puppeteer = require('puppeteer');

(async () => {
  const browser = await puppeteer.launch({
    headless: 'false',
  });
  const page = await browser.newPage();

  await page.goto(
    // small file
    'https://docs.google.com/document/d/1z7yne3yIhdsx4GCW_LwOiqS8_npPC1SvrwhiZOO8cwo/edit',
    // huge file
    // 'https://docs.google.com/document/d/1pv3GjLOAWtT7oeeIunBfzMk2A3hEPqa8B5qK4rtLG0Y/edit?usp=sharing',
    {
      waitUntil: 'networkidle2',
      timeout: 1000 * 60 * 5,
    }
  );
  await page.emulateMediaType('print');

  await page.pdf({
    path: '1_pdf.pdf',
  });
  await page.screenshot({ path: 'before_2_pdf.png' });
  await page.pdf({
    path: '2_pdf.pdf',
  });
  await page.close();
  await browser.close();
})();
```

### Error string

no error

### Puppeteer version

20.5.0

### Node version

16.14.0

### Package manager

npm

### Package manager version

6.14.15

### Operating system

macOS

**Comment by OrKoN** [comment]
Able to repro. It is a bug in Chrome, most likely, and we need to report it to crbug.com.

**Comment by a-apiko** [comment]
@OrKoN, yeah, I expected it. Will create report later. What do you think about that workaround? Have any idea why it works this way or what events or smth triggered by that flow?

**Comment by a-apiko** [comment]
link to crbug - https://bugs.chromium.org/p/chromium/issues/detail?id=1452105
